### PR TITLE
Remove unused ctap module (capsules/src/usb/ctap.rs)

### PR DESCRIPTION
**Pull request by liamjm**
OpenSK has 2 CTAP modules in third_party/tock:
a) capsules/src/usb/ctap.rs is from tock upstream (https://docs.tockos.org/src/capsules/ctap.rs.html)
b) capsules/src/usb/usbc_ctap_hid.rs which is created via a patch (https://github.com/google/OpenSK/blob/develop/patches/tock/02-usb.patch)

This PR removes the tock-supplied module and related references to it. This clears up the code and makes changes to be made without maintaining compatility with unused code. Merging both modules together is a job for another time.

This work is motivated by #446, where we'll need to supply multiple HID reports to the usb stack.

### CODE REVIEW

**Comment by liamjm** [comment]
@kaczmarczyck PTAL

File: patches/tock/08-remove-unused-ctap-module.patch
Comment (ia0) [inline patches/tock/08-remove-unused-ctap-module.patch:1]: For consistency with the other patches, everything before this line should not be present (as well as the signature at the end of the file). You can fix this by using the `./maintainers/patches` script. Simply run the following from your branch:

```
./maintainers/patches apply
./maintainers/patches save
./maintainers/patches restore
```

**Review by ia0 — COMMENTED** [review]
LGTM, but I'll let @kaczmarczyck take a look too.

**Review by kaczmarczyck — DISMISSED** [review]
Minus @ia0 's comment.

**Comment by liamjm** [comment]
> Minus @ia0 's comment.
Fixed.
PTAL

**Review by kaczmarczyck — APPROVED** [review]

**Pull request merged**
